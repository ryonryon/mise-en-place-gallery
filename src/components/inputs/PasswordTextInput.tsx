import styled from "@emotion/styled";
import * as React from "react";

import HideIcon from "../../assets/hide.svg";
import ShowIcon from "../../assets/show.svg";
import InputType from "./input-type";
import { secondary } from "../../constants/colors";

interface Props extends React.InputHTMLAttributes<HTMLInputElement> {
  showHideIcon: boolean;
}

const PasswordTextInput = ({ showHideIcon, ...rest }: Props) => {
  const [imageName, setImageName] = React.useState("show");
  const [inputType, setInputType] = React.useState(InputType.PASSWORD);

  const handleClick = () => {
    setImageName(prevValue => (prevValue === "hide" ? "show" : "hide"));
    setInputType(prevValue => (prevValue === InputType.PASSWORD ? InputType.TEXT : InputType.PASSWORD));
  };

  const PasswordDiv = styled.div`
    display: inline-block;
    width: 70%;
    margin: 16px;

    & > label {
      display: inline-block;
      float: right;
      cursor: pointer;
    }

    & > label > img {
      display: ${showHideIcon ? "inline" : "none"};
      height: 12px;
      wieght: 12px;
    }

    & > input {
      height: 24px;
      width: 100%;
      padding: 4px;
      border: 1px solid ${secondary};
      color: ${secondary};
      border-radius: 4px;

      &::placeholder {
        color: ${secondary};
      }
    }
  `;

  return (
    <PasswordDiv>
      <label htmlFor="password-input" onClick={handleClick}>
        <img src={imageName === "hide" ? HideIcon : ShowIcon} />
      </label>
      <input type={inputType} name="password-input" {...rest} />
    </PasswordDiv>
  );
};

PasswordTextInput.defaultProps = {
  showHideIcon: false
};

export default PasswordTextInput;
