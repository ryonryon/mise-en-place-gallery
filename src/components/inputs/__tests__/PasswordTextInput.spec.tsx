import * as React from "react";
import * as renderer from "react-test-renderer";

import PasswordTextInput from "../PasswordTextInput";

describe("PasswordTextInput test", () => {
  it("tests PasswordTextInput renders correctly ", () => {
    const passwordTextInput = renderer.create(<PasswordTextInput />).toJSON();
    expect(passwordTextInput).toMatchSnapshot();
  });
});
