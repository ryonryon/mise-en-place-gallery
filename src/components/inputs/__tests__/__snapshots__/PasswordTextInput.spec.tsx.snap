// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordTextInput test tests PasswordTextInput renders correctly  1`] = `
<div
  className="css-149i2wz"
>
  <label
    htmlFor="password-input"
    onClick={[Function]}
  >
    <img />
  </label>
  <input
    name="password-input"
    type="password"
  />
</div>
`;
